### PR TITLE
[BEAM-1340,BEAM-1345] Vital moves of user-facing stuff out of util

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
run dataflow validatesrunner

File: sdks/java/core/src/main/java/org/apache/beam/sdk/state/StateBinder.java [hunk 1]
```diff
@@ -17,14 +17,19 @@
...
+import org.apache.beam.sdk.annotations.Internal;
 import org.apache.beam.sdk.coders.Coder;
 import org.apache.beam.sdk.transforms.Combine;
 import org.apache.beam.sdk.transforms.CombineWithContext;
 import org.apache.beam.sdk.transforms.windowing.TimestampCombiner;
 
 /**
- * Visitor for binding a {@link StateSpec} and to the associated {@link State}.
+ * For internal use only; no backwards-compatibility guarantees.
+ * <b><i>For internal use only; no backwards-compatibility guarantees.</i></b>
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/state/StateBinder.java]: dupe text?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/state/StateContext.java [hunk 2]
```diff
@@ -17,13 +17,16 @@
...
 
+import org.apache.beam.sdk.annotations.Internal;
 import org.apache.beam.sdk.options.PipelineOptions;
 import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
 import org.apache.beam.sdk.values.PCollectionView;
 
 /**
- * Information accessible the state API.
+ * For internal use only; no backwards-compatibility guarantees.
+ * <b><i>For internal use only; no backwards-compatibility guarantees.</i></b>
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/state/StateContext.java]: dupe text?

Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/state/StateContext.java]: there are more. happy for you to decide whether you want warning + old javadoc or just warning , which is currently slightly inconsistent. Up to you completely perccase.

**Comment by kennknowles** [comment]
Since this is already pretty heavily covered and blocking @tgroh I'm going to assume the risk and merge with the internal and personal integration testing I've done. There is no substantive code change here anyhow.

I will follow up with the javadoc improvements.

**Pull request merged**